ledger.go

```diff
@@ -0,0 +1,337 @@
+package ledger
+
+import (
+	"fmt"
+)
+
+const (
+	// Spacemesh Application Identifier
+	cCLA = 0x30
+	// Max length of APDU packet supported by Ledger device
+	cMaxPacketLength = 240
+
+	// Instruction code for GetVersion command
+	cInsGetVersion = 0x00
+	// Instruction code for GetExtPublicKey command
+	cInsGetExtPublicKey = 0x10
+	// Instruction code for GetAddress command
+	cInsGetAddress = 0x11
+	// Instruction code for SignTx command
+	cInsSignTx = 0x20
+
+	// Parameter 1 is unused
+	cP1Unused = 0x00
+	// Request type for GetAddress command: returning address to host
+	cP1Return = 0x01
+	// Request type for GetAddress command: displaying address on the screen
+	cP1Display = 0x02
+	// Request Flag for SignTx command: data has transaction header
+	cP1HasHeader = 0x01
+	// Request Flag for SignTx command: data has transaction data
+	cP1HasData = 0x02
+	// Request Flag for SignTx command: last request
+	cP1IsLast = 0x04
+
+	// Parameter 2 is unused
+	cP2Unused = 0x00
+)
+
+// Ledger struct
+type Ledger struct {
+	hid IHidDevice
+}
+
+// Version struct
+type Version struct {
+	Major byte
+	Minor byte
+	Patch byte
+	Flags byte
+}
+
+// ExtendedPublicKey struct
+type ExtendedPublicKey struct {
+	PublicKey []byte
+	ChainCode []byte
+}
+
+// BipPath type
+type BipPath []uint32
+
+// Extract return code from response.
+// param {[]byte} Response data
+// return {[]byte} Response data without return code
+// return {uint32} Error code.
+//
+func stripRetcodeFromResponse(response []byte) ([]byte, uint32) {
+	L := len(response)
+	if L < 2 {
+		return nil, 0
+	}
+	if response[L-2] != 0x90 || response[L-1] != 0x00 { // OK code 0x9000
+		return nil, (uint32(response[L-2]) << 8) + uint32(response[L-1])
```

Comment:
```suggestion
		return nil, binary.BigEndian.Uint32(response[L-2:])
```
Your code is correct, but harder to read/interpret.

Note: you have to import `"encoding/binary"`.